src/plugins/intel_cpu/src/nodes/interpolate.cpp

```diff
@@ -2094,1162 +741,8 @@ void Interpolate::execute(dnnl::stream strm) {
-                NNPlanar(in_ptr_, out_ptr_, post_ops_data_, N, C, ID, IH, IW, OD, OH, OW);
-            } else {
-                NNCGathered(in_ptr_, out_ptr_, post_ops_data_, N, C, ID, IH, IW, OD, OH, OW);
-            }
-            break;
-        }
-        case InterpolateMode::linear_onnx: {
-            if (configured_for_layout == InterpolateLayoutType::planar) {
-                linearOnnxPlanar(in_ptr_, out_ptr_, post_ops_data_, N, C, ID, IH, IW, OD, OH, OW);
-            } else {
-                linearOnnxCGathered(in_ptr_, out_ptr_, post_ops_data_, N, C, ID, IH, IW, OD, OH, OW);
-            }
-            break;
-        }
-        case InterpolateMode::cubic: {
-            if (configured_for_layout == InterpolateLayoutType::planar) {
-                cubicPlanar(in_ptr_, out_ptr_, post_ops_data_, N, C, IH, IW, OH, OW);
-            } else {
-                cubicCGathered(in_ptr_, out_ptr_, post_ops_data_, N, C, IH, IW, OH, OW);
-            }
-            break;
-        }
-        default: {
-            IE_THROW() << "InterpolateJitExecutor has unsupported interpolate mode: " << mode;
-        }
-    }
-}
-
-void Interpolate::InterpolateRefExecutor::exec(const uint8_t *in_ptr_, uint8_t *out_ptr_, const void *post_ops_data_) {
-    size_t N = srcDimPad5d[0], C = srcDimPad5d[1], ID = srcDimPad5d[2], IH = srcDimPad5d[3], IW = srcDimPad5d[4];
-    size_t OD = dstDim5d[2], OH = dstDim5d[3], OW = dstDim5d[4];
-
-    switch (mode) {
-        case InterpolateMode::nearest: {
-            NNRef(in_ptr_, out_ptr_, N, C, ID, IH, IW, OD, OH, OW);
-            break;
-        }
-        case InterpolateMode::linear_onnx: {
-            linearOnnxRef(in_ptr_, out_ptr_, N, C, ID, IH, IW, OD, OH, OW);
-            break;
-        }
-        case InterpolateMode::cubic: {
-            cubicRef(in_ptr_, out_ptr_, N, C, IH, IW, OH, OW);
-            break;
-        }
-        case InterpolateMode::linear: {
-            float fz = (dataRank == 5) ? dataScales[dataRank - 3] : 1.f;
-            float fy = dataScales[dataRank - 2];
-            float fx = dataScales[dataRank - 1];
-
-            bool isDownsample = (fx < 1.f) || (fy < 1.f) || (fz < 1.f);
-            int kernel_width = 2;
-            linearInterpolation(in_ptr_, out_ptr_, N, C, ID, IH, IW, fx, fy, fz, OD, OH, OW, kernel_width, isDownsample && antialias);
-            break;
-        }
-        default: {
-            IE_THROW() << "Interpolate layer has unsupported interpolate mode: " << mode;
-        }
-    }
-}
-
-size_t Interpolate::getSpatialDimsNum(const Dim rank) {
-    switch (rank) {
-        case 1:
-        case 3:
-            return 1;
-        case 2:
-        case 4:
-            return 2;
-        case 5:
-            return 3;
-        default:
-            IE_THROW() << "Can't define number spatial";
-    }
+    srcMemPtr->setDataHandle(const_cast<void *>(reinterpret_cast<const void *>(src_data)));
```

Comment:
That's not legal way to reset source memory. 
I would propose the following:
- Executors still should work with pure memory objects w/o any modifications here (like for other nodes)
- Computations above (for case hasPad=true) should be moved into executors as well. You can put in in base InterpolateExecutor class and reuse between all executors calling it inside exec() method.

Reply:
corrected